### PR TITLE
Use IAM role instead of AWS credentials

### PR DESCRIPTION
**Pull request by iandelahorne**
Use an IAM role to authenticate to redshift instead of using AWS
credentials.

INFRA-64

See docs at https://docs.aws.amazon.com/redshift/latest/dg/copy-parameters-authorization.html

### CODE REVIEW

**Comment by iandelahorne** [comment]
This will probably have to go out in 2 phases in ansible, where we add the IAM role to the JSON configs, deploy, then merge this PR and deploy

**Comment by iandelahorne** [comment]
(Side note I have no idea how to test this)

File: src/main/java/com/patreon/euphrates/Config.java [hunk 1]
```diff
@@ -114,10 +115,12 @@ public Table(
     public S3(
                @JsonProperty("bucket") String bucket,
                @JsonProperty("region") String region,
+               @JsonProperty("accessKey") String iamRole,
```
Comment (kennydo) [inline src/main/java/com/patreon/euphrates/Config.java]: Should the Jon property be "iam role" as well?

**Review by kennydo — COMMENTED** [review]
To test out, I think you can look at the euhorates crons and manually run them and see if they fail. Josh would know best though

File: src/main/java/com/patreon/euphrates/Config.java (hunk 1 above)
Comment (iandelahorne) [reply src/main/java/com/patreon/euphrates/Config.java]: Oops!

**Comment by iandelahorne** [comment]
Is it safe to remove the accessKey / secretKey properties if we're not using them, or will something break?

File: src/main/java/com/patreon/euphrates/Redshift.java [hunk 2]
```diff
@@ -55,9 +55,8 @@ public void copyManifestPath(Config.Table table, String manifestPath) {
         String.format("COPY %s.%s\n", config.redshift.schema, Util.tempTable(table))
           + String.format("FROM 's3://%s/%s'\n", config.s3.bucket, manifestPath)
           + String.format(
-          "CREDENTIALS 'aws_access_key_id=%s;aws_secret_access_key=%s'\n",
```
Comment (joshbuddy) [inline src/main/java/com/patreon/euphrates/Redshift.java:58]: lets remove the accesskey and secretkey parts from the config too, and make iam role the only way to set things up if we're gonna go with this approach

Comment (iandelahorne) [reply src/main/java/com/patreon/euphrates/Redshift.java:58]: :+1:

**Comment by iandelahorne** [comment]
Is this safe to test on any table, or should i create one first?

**Review by joshbuddy — APPROVED** [review]

**Pull request merged**